src/screens/AuthScreen/index.js

```diff
@@ -1,82 +1,87 @@
-import React from 'react';
-import { ActivityIndicator, AsyncStorage, TextInput, View } from 'react-native';
-import { Button, Login, Screen } from '../../components';
+import React, { Component } from 'react';
+import {
+  ActivityIndicator,
+  Alert,
+  AppRegistry,
+  AsyncStorage,
+  Platform,
+  Text,
+  View
+} from 'react-native';
+import { Button, Screen } from '../../components';
+import Auth0 from 'react-native-auth0';
 import theme from '../../theme';
 import styles from './styles';
 
-export default class AuthScreen extends React.Component {
+var credentials = require('./auth0-credentials');
+const auth0 = new Auth0(credentials);
 
-  constructor(props){
+export default class AuthScreen extends React.Component {
+  constructor(props) {
     super(props);
+    this.getExisting()
     this.state = {
-      error: undefined,
-      username: '',
-      isAuthenticating: false
-    }
-  }
-
-  handleAuthenicationError = (err) => {
-    this.setState({
-      error: 'Invalid username or password',
-      isAuthenticating: false
-    });
+      jwt: null,
+      isAuthenticating: true
+    };
   }
 
-  handleAuthenicationSuccess = (username, jwt) => {
-    this.setState({
-      error: undefined,
-      isAuthenticating: false
-    });
-    // Store session information and proceed to `Scan` screen
-    const appState = JSON.stringify({ username, jwt });
-    AsyncStorage.setItem(this.props.storageKey, appState)
-      .then(() => {
-        // Navigate to the scan screen
+  getExisting = () => {
+    AsyncStorage.getItem(this.props.storageKey).then((value) => {
+      const { jwt } = JSON.parse(value)
+      this.setState({ jwt, isAuthenticating: false })
+      if (jwt) {
         this.props.navigation.navigate('Scan');
-      });
+      } else {
+        this.onLogin()
+      }
+    })
   }
 
-  signIn = (username, password) => {
-    this.setState({
-      isAuthenticating: true,
-      username
-    });
-    this.props.databroker
-      .authenticate(username, password)
-      .then(this.handleAuthenicationSuccess.bind(this, username))
-      .catch(this.handleAuthenicationError)
-  }
-
-  componentDidMount(){
-    if(this.props.navigation.state.params){
-      console.log('current username!', this.props.navigation.state.params.username);
-      this.setState({ username: this.props.navigation.state.params.username})
-    }
-  }
-
-  render(){
-    const { error, username, isAuthenticating } = this.state;
-    const loadingDisplay = isAuthenticating ? 'flex' : 'none';
+  onLogin = () => {
```

Comment:
Why did you remove the syntax that already handled the promise? All you had to change was the `this.props.databroker.authenticate` to -> `auth0.webAuth.authorize(...)`

Reply:
fixed